// @moderation-api/sdk/cjs/generated/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueActionsService = exports.ModerateService = exports.AccountService = exports.OpenAPI = exports.CancelError = exports.CancelablePromise = exports.BaseHttpRequest = exports.ApiError = exports.ModerationAPI = void 0;
/* generated using openapi-typescript-codegen -- do no edit */
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */
var ModerationAPI_1 = require("./ModerationAPI");
Object.defineProperty(exports, "ModerationAPI", { enumerable: true, get: function () { return ModerationAPI_1.ModerationAPI; } });
var ApiError_1 = require("./core/ApiError");
Object.defineProperty(exports, "ApiError", { enumerable: true, get: function () { return ApiError_1.ApiError; } });
var BaseHttpRequest_1 = require("./core/BaseHttpRequest");
Object.defineProperty(exports, "BaseHttpRequest", { enumerable: true, get: function () { return BaseHttpRequest_1.BaseHttpRequest; } });
var CancelablePromise_1 = require("./core/CancelablePromise");
Object.defineProperty(exports, "CancelablePromise", { enumerable: true, get: function () { return CancelablePromise_1.CancelablePromise; } });
Object.defineProperty(exports, "CancelError", { enumerable: true, get: function () { return CancelablePromise_1.CancelError; } });
var OpenAPI_1 = require("./core/OpenAPI");
Object.defineProperty(exports, "OpenAPI", { enumerable: true, get: function () { return OpenAPI_1.OpenAPI; } });
var AccountService_1 = require("./services/AccountService");
Object.defineProperty(exports, "AccountService", { enumerable: true, get: function () { return AccountService_1.AccountService; } });
var ModerateService_1 = require("./services/ModerateService");
Object.defineProperty(exports, "ModerateService", { enumerable: true, get: function () { return ModerateService_1.ModerateService; } });
var QueueActionsService_1 = require("./services/QueueActionsService");
Object.defineProperty(exports, "QueueActionsService", { enumerable: true, get: function () { return QueueActionsService_1.QueueActionsService; } });
