// @moderation-api/sdk/cjs/generated/ModerationAPI.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModerationAPI = void 0;
const AxiosHttpRequest_1 = require("./core/AxiosHttpRequest");
const AccountService_1 = require("./services/AccountService");
const ModerateService_1 = require("./services/ModerateService");
const QueueActionsService_1 = require("./services/QueueActionsService");
class ModerationAPI {
    constructor(config, HttpRequest = AxiosHttpRequest_1.AxiosHttpRequest) {
        var _a, _b, _c, _d;
        this.request = new HttpRequest({
            BASE: (_a = config === null || config === void 0 ? void 0 : config.BASE) !== null && _a !== void 0 ? _a : 'https://moderationapi.com/api/v1',
            VERSION: (_b = config === null || config === void 0 ? void 0 : config.VERSION) !== null && _b !== void 0 ? _b : '1.0.0',
            WITH_CREDENTIALS: (_c = config === null || config === void 0 ? void 0 : config.WITH_CREDENTIALS) !== null && _c !== void 0 ? _c : false,
            CREDENTIALS: (_d = config === null || config === void 0 ? void 0 : config.CREDENTIALS) !== null && _d !== void 0 ? _d : 'include',
            TOKEN: config === null || config === void 0 ? void 0 : config.TOKEN,
            USERNAME: config === null || config === void 0 ? void 0 : config.USERNAME,
            PASSWORD: config === null || config === void 0 ? void 0 : config.PASSWORD,
            HEADERS: config === null || config === void 0 ? void 0 : config.HEADERS,
            ENCODE_PATH: config === null || config === void 0 ? void 0 : config.ENCODE_PATH,
        });
        this.account = new AccountService_1.AccountService(this.request);
        this.moderate = new ModerateService_1.ModerateService(this.request);
        this.queueActions = new QueueActionsService_1.QueueActionsService(this.request);
    }
}
exports.ModerationAPI = ModerationAPI;
